### PR TITLE
0.8 preview

### PR DESCRIPTION
**Pull request by atotic**
Incomplete port.
PORT0.8.txt lists status of all current ports

### CODE REVIEW

File: bower.json [hunk 1]
```diff
@@ -25,6 +25,10 @@
     "apis"
   ],
   "dependencies": {
-    "core-shared-lib": "Polymer/core-shared-lib#^0.5.5"
+    "polymer": "Polymer/polymer#0.8-preview",
+    "iron-jsonp-library": "PolymerElements/iron-jsonp-library#^0.8.0",
+  },
+  "devDependencies": {
+    "iron-doc-viewer": "PolymerElements/iron-doc-viewer#^0.8.6"
```
Comment (ebidel) [inline bower.json]: 0.8.8

File: bower.json [hunk 2]
```diff
@@ -25,6 +25,10 @@
     "apis"
   ],
   "dependencies": {
-    "core-shared-lib": "Polymer/core-shared-lib#^0.5.5"
+    "polymer": "Polymer/polymer#0.8-preview",
+    "iron-jsonp-library": "PolymerElements/iron-jsonp-library#^0.8.0",
```
Comment (ebidel) [inline bower.json]: Has this been tagged?

File: demo.html [hunk 3]
```diff
@@ -4,40 +4,44 @@
...
-  <google-realtime-api></google-realtime-api>
-  <google-youtube-api></google-youtube-api>
+  <template  id="t" is="x-autobind">
 
-  <div></div>
+    <google-client-loader id="shortener" 
+      name="urlshortener"
+      version="v1"
+      on-google-api-load="loadedShortener"></google-client-loader>
+    <google-js-api        on-js-api-load="loaded"></google-js-api>
```
Comment (ebidel) [inline demo.html]: please remove the aligned spacing.

File: google-client-loader.html [hunk 4]
```diff
@@ -11,72 +11,15 @@
 <link rel="import" href="google-js-api.html">
 
 <!--
-Dynamically loads the Google Client JavaScript API, firing the `api-load` event when ready.
+Element for loading a specific client Google API with the Javascript client library.
```
Comment (ebidel) [inline google-client-loader.html]: JavaScript

File: google-client-loader.html [hunk 5]
```diff
@@ -91,79 +34,85 @@
...
 
 @element google-api-loader
-@extends google-client-api
 @homepage https://googlewebcomponents.github.io/google-apis
 -->
-<polymer-element name="google-api-loader"
-                 attributes="name version appId root"
-                 extends="google-client-api"
-                 on-api-load="{{doneLoadingClient}}">
+<dom-module id='google-client-loader'>
```
Comment (ebidel) [inline google-client-loader.html]: double quotes for attrs

File: google-client-loader.html [hunk 6]
```diff
@@ -91,79 +34,85 @@
...
+         */
+        appId: String,
+
+        /**
+         * Root URL where to load the API from, e.g. 'http://host/apis'.
+         * For App Engine dev server this would be something like:
+         * 'http://localhost:8080/_ah/api'.
+         * Overrides 'appId' if both are specified.
+         * @attribute root
+         * @type string
```
Comment (ebidel) [inline google-client-loader.html]: @type String

File: google-client-loader.html [hunk 7]
```diff
@@ -91,79 +34,85 @@
...
+         * @attribute name
+         * @type string
+         * @required
+         */
+        name: String,
+
+        /**
+         * Version of the API to load, e.g. 'v1'.
+         * @attribute version
+         * @type string
```
Comment (ebidel) [inline google-client-loader.html]: @type String

File: google-client-loader.html [hunk 8]
```diff
@@ -91,79 +34,85 @@
...
-      root: null,
+      properties: {
+        /**
+         * Name of the API to load, e.g. 'urlshortener'.
+         *
+         * You can find the full list of APIs on the
+         * <a href="https://developers.google.com/apis-explorer"> Google APIs
+         * Explorer</a>.
+         * @attribute name
+         * @type string
```
Comment (ebidel) [inline google-client-loader.html]: @type String

File: google-client-loader.html [hunk 9]
```diff
@@ -91,79 +34,85 @@
...
-       */
-      root: null,
+      properties: {
+        /**
+         * Name of the API to load, e.g. 'urlshortener'.
+         *
+         * You can find the full list of APIs on the
+         * <a href="https://developers.google.com/apis-explorer"> Google APIs
+         * Explorer</a>.
+         * @attribute name
```
Comment (ebidel) [inline google-client-loader.html]: all of these are properties now, not attrs

File: google-client-loader.html [hunk 10]
```diff
@@ -195,12 +144,27 @@
         }
       },
 
+      /**
+       * Wrapper for `gapi.auth`.
+       * @property auth
+       * @type gapi.auth
+       */
+      get auth() {
```
Comment (ebidel) [inline google-client-loader.html]: does this still work in 0.8? Should we do `value: {return gap.auth;}` instead?

File: google-client-loader.html [hunk 11]
```diff
@@ -195,12 +144,27 @@
...
+        return gapi.auth;
+      },
+
+      loadClient: function() {
+        gapi.load("client", function() {
```
Comment (ebidel) [inline google-client-loader.html]: single quotes

Comment (ebidel) [inline google-client-loader.html]: simplify to: `gapi.load('client', this.doneLoadingClient.bind(this));`

File: google-maps-api.html [hunk 12]
```diff
@@ -36,95 +35,130 @@
...
+      mapsUrl: {
+        type: String,
+        value: 'https://maps.googleapis.com/maps/api/js?callback=%%callback%%'
+      },
+
+      /**
+       * A Maps API key. To obtain an API key, see developers.google.com/maps/documentation/javascript/tutorial#api_key.
+       *
+       * @attribute apiKey
+       * @type string
```
Comment (ebidel) [inline google-maps-api.html]: `@type String`. please change throughout.

File: google-maps-api.html [hunk 13]
```diff
@@ -36,95 +35,130 @@
...
+       */
+      apiKey: {
+        type: String,
+        value: ''
```
Comment (ebidel) [inline google-maps-api.html]: does hydrolysis support default values?

File: google-maps-api.html [hunk 14]
```diff
@@ -36,95 +35,130 @@
...
+       * The libraries to load with this map. Defaults to "places". For more information
+       * see https://developers.google.com/maps/documentation/javascript/libraries.
+       *
+       * @attribute libraries
+       * @type string
+       * @default "places"
+       */
+      libraries: {
+        type: String,
+        value: "places"
```
Comment (ebidel) [inline google-maps-api.html]: singles quotes

**Comment by ebidel** [comment]
Where is `Polymer.IronJsonpLibraryBehavior`. Am I missing that. 

This will also need a rebase, not sure why :\

File: bower.json (hunk 1 above)
Comment (atotic) [inline bower.json]: fixed

File: bower.json (hunk 2 above)
Comment (atotic) [inline bower.json]: What do you mean by "tagged". Its bower.json has a 0.8.0

File: demo.html (hunk 3 above)
Comment (atotic) [inline demo.html]: Mwah, readability looses. Done.

File: google-client-loader.html (hunk 4 above)
Comment (atotic) [inline google-client-loader.html]: done

File: google-client-loader.html (hunk 5 above)
Comment (atotic) [inline google-client-loader.html]: fixed.

File: google-client-loader.html (hunk 6 above)
Comment (atotic) [inline google-client-loader.html]: All the @type annotations will go away, as iron-doc infers them from source. Not fixed.

File: google-client-loader.html (hunk 7 above)
Comment (atotic) [inline google-client-loader.html]: Again, @type will go away.

File: google-client-loader.html (hunk 9 above)
Comment (atotic) [inline google-client-loader.html]: In general, our documentation looks bad in iron-doc, that'll be a separate "make docs look pretty" pass when iron-doc is stable.

File: google-client-loader.html (hunk 10 above)
Comment (atotic) [inline google-client-loader.html]: It works, I have no preference. The way it looks in the docs might swing the vote.

File: google-client-loader.html (hunk 11 above)
Comment (atotic) [inline google-client-loader.html]: done

File: google-maps-api.html (hunk 13 above)
Comment (atotic) [inline google-maps-api.html]: not sure, but I expect it to.

File: google-maps-api.html (hunk 14 above)
Comment (atotic) [inline google-maps-api.html]: fixed.

**Comment by atotic** [comment]
Polymer.IronJsonpLibraryBehavior is at https://github.com/PolymerElements/iron-jsonp-library

File: bower.json (hunk 2 above)
Comment (ebidel) [inline bower.json]: It needs a 0.8.0 tag (https://github.com/PolymerElements/iron-jsonp-library/releases) for bower to be able to pull that version.

File: google-client-loader.html (hunk 9 above)
Comment (ebidel) [inline google-client-loader.html]: Ack.

File: google-client-loader.html (hunk 7 above)
Comment (ebidel) [inline google-client-loader.html]: Yea, we should probably just remove these then. But I'm ok with followups for doc cleanup

File: google-client-loader.html (hunk 10 above)
Comment (ebidel) [inline google-client-loader.html]: getters were more of an issue in 0.5 when Object.observer() wasn't present. Think it's fine to leave, but I'll leave it up to you.

**Comment by ebidel** [comment]
Ack on `Polymer.IronJsonpLibraryBehavior `

**Comment by ebidel** [comment]
LGTM. Can you rebase so this can be merged?

File: bower.json (hunk 2 above)
Comment (atotic) [inline bower.json]: tagged

Comment (ebidel) [inline bower.json]: Thanks

**Comment by atotic** [comment]
Rebase I can't do because my git-fu is not up to par. Would love to get instruction.

**Pull request merged**